pvlib/shading.py

```diff
@@ -0,0 +1,193 @@
+    phi = np.arctan(numerator / denominator)
+    return np.degrees(phi)
+
+
+def masking_angle_passias(surface_tilt, gcr):
+    r"""
+    The average masking angle over the slant height of a row.
+
+    The masking angle is the angle from horizontal where the sky dome is
+    blocked by the row in front. The masking angle is larger near the lower
+    edge of a row than near the upper edge. This function calculates the
+    average masking angle as described in [1]_.
+
+    Parameters
+    ----------
+    surface_tilt : numeric
+        Panel tilt from horizontal [degrees].
+
+    gcr : float
+        The ground coverage ratio of the array [unitless].
+
+    Returns
+    ----------
+    mask_angle : numeric
+        Average angle from horizontal where diffuse light is blocked by the
+        preceding row [degrees].
+
+    See Also
+    --------
+    masking_angle
+    sky_diffuse_passias
+
+    Notes
+    -----
+    The pvlib-python authors believe that Eqn. 9 in [1]_ is incorrect.
+    Here we use an independent equation.  First, Eqn. 8 is non-dimensionalized
+    (recasting in terms of GCR):
+
+    .. math::
+
+        \psi(z') = \arctan \left [
+            \frac{(1 - z') \sin \beta}
+                 {\mathrm{GCR}^{-1} + (z' - 1) \cos \beta}
+        \right ]
+
+    Where :math:`GCR = B/C` and :math:`z' = z/B`. The average masking angle
+    :math:`\overline{\psi} = \int_0^1 \psi(z') \mathrm{d}z'` is then
+    evaluated symbolically using Maxima (using :math:`X = 1/\mathrm{GCR}`):
+
+    .. code-block:: none
+
+        load(scifac)    /* for the gcfac function */
+        assume(X>0, cos(beta)>0, cos(beta)-X<0);   /* X is 1/GCR */
+        gcfac(integrate(atan((1-z)*sin(beta)/(X+(z-1)*cos(beta))), z, 0, 1))
+
+    This yields the equation implemented by this function:
+
+    .. math::
+
+        \overline{\psi} = \
+            &-\frac{X}{2} \sin\beta \log | 2 X \cos\beta - (X^2 + 1)| \\
+            &+ (X \cos\beta - 1) \arctan \frac{X \cos\beta - 1}{X \sin\beta} \\
+            &+ (1 - X \cos\beta) \arctan \frac{\cos\beta}{\sin\beta} \\
+            &+ X \log X \sin\beta
+
+    The pvlib-python authors have validated this equation against numerical
+    integration of :math:`\overline{\psi} = \int_0^1 \psi(z') \mathrm{d}z'`.
+
+    References
+    ----------
+    .. [1] D. Passias and B. Källbäck, "Shading effects in rows of solar cell
+       panels", Solar Cells, Volume 11, Pages 281-291.  1984.
+       DOI: 10.1016/0379-6787(84)90017-6
+    """
+    # wrap it in an array so that division by zero is handled well
+    beta = np.radians(np.array(surface_tilt))
+    sin_b = np.sin(beta)
+    cos_b = np.cos(beta)
+    X = 1/gcr
+
+    with np.errstate(divide='ignore', invalid='ignore'):  # ignore beta=0
+        term1 = -X * sin_b * np.log(np.abs(2 * X * cos_b - (X**2 + 1))) / 2
+        term2 = (X * cos_b - 1) * np.arctan((X * cos_b - 1) / (X * sin_b))
+        term3 = (1 - X * cos_b) * np.arctan(cos_b / sin_b)
```

Comment:
@kanderso-nrel  I believe that `arctan(cotangent(beta)) = beta-pi/2` right?

Reply:
Yeah it's silly to have `arctan(cot(x))`.  The problem is that I wanted the output of that Maxima snippet, the latex equation, and the python code to all be identical but I couldn't figure out what assumptions I needed to give Maxima to simplify that term.  I figured the inefficiency of that python term was negligible compared with making the connections as obvious as possible, but maybe there's a clearer way to tie the three pieces together?

Reply:
See [trig identities: inverse trig funcs](https://en.wikipedia.org/wiki/List_of_trigonometric_identities#Inverse_trigonometric_functions)

    arctan x + arccot x = pi / 2
    arctan x = pi / 2 - arccot x

so

    arctan(cot(beta)) = pi / 2 - arccot(cot(beta)) = pi / 2 - beta

QED

Reply:
Don't sweat it. Let's merge.